plugins/providers/hyperv/scripts/import_vm_vmcx.ps1

```diff
@@ -0,0 +1,167 @@
+﻿Param(
+    [Parameter(Mandatory=$true)]
+    [string]$vm_config_file,
+    [Parameter(Mandatory=$true)]
+    [string]$image_path,
+
+    [string]$switchname=$null,
+    [string]$memory=$null,
+    [string]$maxmemory=$null,   
+    [string]$cpus=$null,
+    [string]$vmname=$null,
+    [string]$auto_start_action=$null,
+    [string]$auto_stop_action=$null
+)
+
+# Include the following modules
+$Dir = Split-Path $script:MyInvocation.MyCommand.Path
+. ([System.IO.Path]::Combine($Dir, "utils\write_messages.ps1"))
+
+# load the config from the vmcx and make a copy for editing, use TMP path so we are sure there is no vhd at the destination
+$vmConfig = (Compare-VM -Copy -Path $vm_config -GenerateNewID -VhdDestinationPath $env:Temp)
```

Comment:
Ok, I might've did an oops with the laat commit there. I'll check when I have a pc available.